### ISSUE TITLE
"\"x-hasura-\" header is expected but not found"

### ISSUE THREAD

**Issue by papacarp** [hasura/graphql-engine#5068]
I have an INSERT custom check with (among other things) the following rule:

![image](https://user-images.githubusercontent.com/11194546/84436873-f9cdd500-abe8-11ea-8a0f-f23edf0bc424.png)

I use the console to provide the auth:
![image](https://user-images.githubusercontent.com/11194546/84436948-208c0b80-abe9-11ea-9384-3abc54b1c8a3.png)

I use a customs namespace:
`HASURA_GRAPHQL_JWT_SECRET: '{"claims_namespace_path":"hasura","jwk_url": "https://cog....`

The JWT has the appropriate headers in 

![image](https://user-images.githubusercontent.com/11194546/84437220-7b256780-abe9-11ea-8b72-dbce5515dee8.png)

The result is:
```
{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "not-found"
      },
      "message": "\"x-hasura-\" header is expected but not found"
    }
  ]
}
```

**Comment by leoalves** [comment]
Hi @papacarp,

I think your custom claims is a string and not a JSON object.  In your `HASURA_GRAPHQL_JWT_SECRET` did you set the `claims_format` to `stringified_json`. 
https://hasura.io/docs/1.0/graphql/manual/auth/authentication/jwt.html#configuring-jwt-mode

**Comment by papacarp** [comment]
Yes, claims format is stringified json.  I should also mention that the platform works fine for everything else.  authentication is not an issue.  its this specific problem with inserts

**Comment by JeremyTsaii** [comment]
Having the same problem with inserts

**Comment by jmarbutt** [comment]
I am having a similar issue when using a header variable in a select constraint, did you ever find a solution for this?

**Comment by jbek7** [comment]
@jmarbutt did you find a solution for this?

**Comment by jmarbutt** [comment]
> @jmarbutt did you find a solution for this?

The issue I found is that I was testing it with the `x-hasura-admin-secret` but when running the app I was using another JWT token. What I found was that you can append headers **with** the admin secret but **not** when using using a JWT token.

**Comment by jbek7** [comment]
I think, when we use `x-hasura-admin-secret`, it doesn't use any permission checking because everything is granted with it. But somehow I'm having this problem with JWT auth too.

**Comment by jmarbutt** [comment]
@jbek7  Can you show the contents of your JWT?
